pkg/runtime/wfengine/workflow.go

```diff
@@ -184,20 +196,62 @@ func (wf *workflowActor) createWorkflowInstance(ctx context.Context, actorID str
 		}
 	}
 
-	// We block (re)creation of existing workflows unless they are in a completed state
-	// Or if they still have any pending activity result awaited.
-	if !created {
-		runtimeState := getRuntimeState(actorID, state)
-		if !runtimeState.IsCompleted() {
-			return fmt.Errorf("an active workflow with ID '%s' already exists", actorID)
-		}
-		if wf.activityResultAwaited.Load() {
-			return fmt.Errorf("a terminated workflow with ID '%s' is already awaiting an activity result", actorID)
+	// orchestration didn't exist and was just created
+	if created {
+		return wf.scheduleWorkflowStart(ctx, actorID, startEvent, state)
+	}
+
+	// orchestration already existed: apply reuse id policy
+	runtimeState := getRuntimeState(actorID, state)
+	runtimeStatus := runtimeState.RuntimeStatus()
+	// if target status doesn't match, fall back to original logic, create instance only if previous one is completed
+	if !isStatusMatch(reuseIDPolicy.GetOperationStatus(), runtimeStatus) {
+		return wf.createIfCompleted(ctx, runtimeState, actorID, state, startEvent)
+	}
+
+	switch reuseIDPolicy.GetAction() {
```

Comment:
The PR LGTM, I just wanted to understand how are we skipping the creation of a new orchestration here, (i.e. the option `SkipIfExists` from https://github.com/microsoft/durabletask-go/issues/42#issue-1966761413)

Reply:
```go
case api.REUSE_ID_ACTION_IGNORE:
		// Log an warning message and ignore creating new instance
		wfLogger.Warnf("Workflow actor '%s': ignoring request to recreate the current workflow instance", actorID)
		return nil
```

So if users choose `REUSE_ID_ACTION_IGNORE` and if there is an existing instance with the same instance ID, then we will enter the above block and just log a message and directly return.

Reply:
my bad, I missed that Ignore is Skip, thanks for clarification